fedify/testing/mock.ts

```diff
@@ -0,0 +1,691 @@
+        return this;
+      },
+      setSharedKeyDispatcher(): InboxListenerSetters<TContextData> {
+        return this;
+      },
+    };
+  }
+
+  async startQueue(
+    contextData: TContextData,
+    _options?: FederationStartQueueOptions,
+  ): Promise<void> {
+    this.contextData = contextData;
+    // not actually starting a queue
+  }
+
+  async processQueuedTask(
+    contextData: TContextData,
+    _message: Message,
+  ): Promise<void> {
+    this.contextData = contextData;
+    // no queue in mock type. process immediately
+  }
+
+  createContext(baseUrl: URL, contextData: TContextData): Context<TContextData>;
+  createContext(
+    request: Request,
+    contextData: TContextData,
+  ): RequestContext<TContextData>;
+  createContext(
+    baseUrlOrRequest: URL | Request,
+    contextData: TContextData,
+  ): Context<TContextData> | RequestContext<TContextData> {
+    const mockFederation = this;
+
+    if (baseUrlOrRequest instanceof Request) {
+      return createRequestContext({
+        url: new URL(baseUrlOrRequest.url),
+        request: baseUrlOrRequest,
+        data: contextData,
+        federation: mockFederation as any,
+        sendActivity: async (_sender, _recipients, activity) => {
+          mockFederation.sentActivities.push(activity);
+        },
+      });
+    } else {
+      return createContext({
+        url: baseUrlOrRequest,
+        data: contextData,
+        federation: mockFederation as any,
+        sendActivity: async (_sender, _recipients, activity) => {
+          mockFederation.sentActivities.push(activity);
+        },
+      });
+    }
+  }
+
+  async fetch(
+    request: Request,
+    options: FederationFetchOptions<TContextData>,
+  ): Promise<Response> {
+    // returning 404 by default
+    if (options.onNotFound) {
+      return options.onNotFound(request);
+    }
+    return new Response("Not Found", { status: 404 });
+  }
+
+  /**
+   * Simulates receiving an activity. This method is specific to the mock
+   * implementation and is used for testing purposes.
+   *
+   * @param activity The activity to receive.
+   * @returns A promise that resolves when the activity has been processed.
+   * @since 1.8.0
+   */
+  async receiveActivity(activity: Activity): Promise<void> {
+    this.receivedActivities.push(activity);
+
+    // Find and execute appropriate inbox listeners
+    const typeName = activity.constructor.name;
+    const listeners = this.inboxListeners.get(typeName) || [];
+
+    for (const listener of listeners) {
+      const context = createInboxContext({
+        data: this.contextData!,
+        federation: this as any,
+      });
+      await listener(context, activity);
+    }
+  }
+
+  /**
+   * Gets all activities that have been sent through this mock federation.
+   * This method is specific to the mock implementation and is used for
+   * testing purposes.
+   *
+   * @returns An array of sent activities.
+   * @since 1.8.0
+   */
+  getSentActivities(): Activity[] {
```

Comment:
Would it be better to make this a getter? Then you could write: `mock.sentActivities.at(0)....` without the extra function call?

Reply:
Recently I got too used to Golang, so I got a habit of making everything into getters 😅
Removed https://github.com/fedify-dev/fedify/pull/283/commits/97cb4c558d7b9cd09a30335909019c0a2ed591fd